assignments/this.js

```diff
@@ -1,26 +1,76 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. Window Binding, is like the forest of trees, not sure which one we are pointing at, so we point at the forest. The this keyword always tries to point to the Window in any place. It works with functions.
+*
+* 2. Implicit Binding,The object that is standing before the dot is what this keyword will be bound to. You use your pointer to point at the exact location where you want to be. This binding is automatic with objects and methods.
+*
+* 3. New Binding, Create a new object using object literal, add properties and methods and return this. A constructor function is a function that returns an object. It is an Object creator. When we call it, instead of the 'this' we use 'new' keyword. yikesss
+*
+* 4. Explicit Binding, We explicitly say to a function what object it should use for this — using functions such as call, apply and bind. It is always for functions only. It tells us where we want to go. We have full control of the 'this'keywrod. CALL, APPLY, BIND - - >  :)
 *
 * write out a code example of each explanation above
 */
+// There is no binding in arrow function. Arrow function don't bind witht he 'this' keyword.If you use arrow functions it will direct you to the forest/global :(
 
 // Principle 1
 
 // code example for Window Binding
+function hello(name){
+    "use strict";  //Don't bind to the global, stay in the scope of this funtion. Preventing us from pointing to the forest. :) 'use strict' prevents use from accidently biniding to the window.
+    return `Hello ${this.name}`;
```

Comment:
You aren't actually referencing anything here, because `this` keyword, can't reference parameters. It can reference some kind of a variable. Either an Object, or an attribute, or a `const variable`, etc.

when they say // code example for Window Binding 
The `Window` is actually an object that houses all the space we write our code in.
So the way you would reference it, is: `return this`